File: Strings/2566. Maximum Difference by Remapping a Digit/2566. Maximum Difference by Remapping a Digit.py
class Solution:
    def minMaxDifference(self, num: int) -> int:
        maxx = minn = str(num)

        i = 0
        while i < len(maxx) and maxx[i] == "9":
            i += 1
        
        if i < len(maxx):
            maxx = maxx.replace(maxx[i], "9")
        minn = minn.replace(minn[0], "0")

        return int(maxx) - int(minn)

# Time: O(n)
# Space: O(1)

File: Math/2566. Maximum Difference by Remapping a Digit/2566. Maximum Difference by Remapping a Digit.js
/**
 * @param {number} num
 * @return {number}
 */
var minMaxDifference = function (num) {
  let maxx = num.toString();
  let minn = num.toString();

  let i = 0;
  while (i < maxx.length && maxx[i] === "9") i++;

  if (i < maxx.length) maxx = maxx.replaceAll(maxx[i], "9");
  minn = minn.replaceAll(minn[0], "0");

  return Number(maxx) - Number(minn);
};

// Time: O(n)
// Space: O(1)
